src/sentry/db/models/manager/base_query_set.py

```diff
@@ -1,15 +1,62 @@
 import abc
-from typing import Any
+from typing import Any, List
 
-from django.db import router
-from django.db.models import QuerySet
+from django.db import connections, router, transaction
+from django.db.models import QuerySet, sql
+
+from sentry.signals import post_update
 
 
 class BaseQuerySet(QuerySet, abc.ABC):
-    # XXX(dcramer): we prefer values_list, but we can't disable values as Django uses it
-    # internally
-    # def values(self, *args, **kwargs):
-    #     raise NotImplementedError('Use ``values_list`` instead [performance].')
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+        self._send_post_update_signal = False
+
+    def enable_post_update_signal(self, enable: bool) -> "BaseQuerySet":
+        """
+        Enables sending a `post_update` signal after this queryset runs an update command. Note that this is less
+        efficient than just running the update. To get the list of group ids affected, we first run the query to
+        fetch the rows we want to update, then run the update using those ids.
+        """
+        qs = self.all()
+        qs._send_post_update_signal = enable
+        return qs
+
+    def _clone(self) -> "BaseQuerySet":
+        qs = super()._clone()  # type: ignore[misc]
+        qs._send_post_update_signal = self._send_post_update_signal
+        return qs
+
+    def update_with_returning(self, returned_fields: List[str], **kwargs):
+        """
+        Copied and modified from `Queryset.update()` to support `RETURNING <returned_fields>`
+        """
+        self._not_support_combined_queries("update")  # type: ignore[attr-defined]
+        assert not self.query.is_sliced, "Cannot update a query once a slice has been taken."
+        self._for_write = True
+        query = self.query.chain(sql.UpdateQuery)
+        query.add_update_values(kwargs)  # type: ignore[attr-defined]
+        # Clear any annotations so that they won't be present in subqueries.
+        query.annotations = {}
+        with transaction.mark_for_rollback_on_error(using=self.db):
+            query_sql, query_params = query.get_compiler(self.db).as_sql()
+            query_sql += f" RETURNING {', '.join(returned_fields)} "
+            using = router.db_for_write(self.model)
+
+            with connections[using].cursor() as cursor:
+                cursor.execute(query_sql, query_params)
+                result_ids = cursor.fetchall()
+        self._result_cache = None
+        return result_ids
+
+    def update(self, **kwargs: Any) -> int:
+        if self._send_post_update_signal:
+            ids = [result[0] for result in self.update_with_returning(["id"], **kwargs)]
```

Comment:
I don't think we actually use anything but the default artificial PK, but just in case this would be more correct for the corner cases, if any: 

```suggestion
            pk_column = self._meta.pk.name  # "id" by default 
            ids = [result[0] for result in self.update_with_returning([pk_column], **kwargs)]
```